romanisim/tests/test_bandpass.py

```diff
@@ -64,10 +72,161 @@ def test_compute_abflux(filter):
+    wave_bin_width = wavedist[1] - wavedist[0]
+
+    # Define spectral features
+    # Linear slope
+    flux_slope = np.array([1, 0.5, 0])
+    # Curve
+    flux_arg = argus.pdf(x=wavedist, chi=1, loc=1, scale=1)
+    # Pedestal 1
+    flux_flat1 = np.array([0.66, 0.66])
+    # Dirac Delta function
+    wavedist_dd = np.linspace(2.13 - 0.001, 2.13 + 0.001, 1000)
+    flux_dd = norm.pdf(wavedist_dd, 2.13, 0.001)
+    # Pedestal 2
+    flux_flat2 = np.array([0.33, 0.33])
+
+    # Array to store the theoretical spectral flux density
+    theo_flux = np.zeros(len(effarea['Wave']))
+
+    # Linear slope from 400nm to 1000nm
+    arg_start = np.where(wavedist.value == 1)[0][0]
+    theo_flux[0:arg_start] = np.arange(start=1, stop=0, step=(-1 / (arg_start)))
+
+    # Define spectral flux array and wavelengths (uneven spacing)
+    total_flux = flux_slope.copy()
+    total_wavedist = np.array([0.4, 0.7, 1])
+
+    # Curve from 1000nm to 2000nm
+    arg_stop = np.where(wavedist.value == 2)[0][0]
+    arg_wavedist = wavedist.value[arg_start + 1:arg_stop]
+    total_flux = np.append(total_flux, flux_arg[np.where(wavedist.value == 1)[0][0] + 1:
+                                                np.where(wavedist.value == 2)[0][0]])
+    total_wavedist = np.append(total_wavedist, arg_wavedist)
+    theo_flux[arg_start:arg_stop + 1] = flux_arg[arg_start:arg_stop + 1]
+
+    # Pedestal from 2000nm to 2130nm
+    dd_loc = np.where(wavedist.value == 2.13)[0][0] + 1
+    total_flux = np.append(total_flux, flux_flat1)
+    total_wavedist = np.append(total_wavedist, np.array([2.0, 2.13 - 0.001]))
+    theo_flux[arg_stop + 1:dd_loc] = flux_flat1[0]
+
+    # Delta function at 2130nm
+    total_flux = np.append(total_flux, flux_dd)
+    total_wavedist = np.append(total_wavedist, np.array(wavedist_dd))
+    theo_flux[dd_loc] = 1 / wave_bin_width.value
+
+    # Pedestal from 2130nm to 2600nm
+    total_flux = np.append(total_flux, flux_flat2)
+    total_wavedist = np.append(total_wavedist, np.array([2.13 + 0.001, 2.6]))
+    theo_flux[dd_loc + 1:] = flux_flat2[0]
+
+    # Rescale both spectra
+    total_flux *= 1.0e-35
+    theo_flux *= 1.0e-35
+
+    # Add spectral flux density units
+    total_flux *= u.erg / (u.s * u.cm ** 2 * u.hertz)
+    theo_flux *= u.erg / (u.s * u.cm ** 2 * u.hertz)
+
+    for filter in IFILTLIST:
+        # Define filter area
+        area = effarea[filter] * u.m ** 2
+
+        # Analytical flux
+        theo_counts = (wave_bin_width * (np.divide(np.multiply(area, theo_flux), wavedist)
+                                         / constants.h.to(u.erg * u.s))).to(1 / u.s)
+
+        # Sum the flux in the filter
+        theo_counts_sum = np.sum(theo_counts)
+
+        # Computed flux
+        gauss_flux = bandpass.compute_count_rate(flux=total_flux, bandpass=filter, wavedist=total_wavedist)
+
+        # Test that proper results (within 4%) are returned for select bands.
+        assert np.isclose(theo_counts_sum.value, gauss_flux, rtol=4.0e-2)
```

Comment:
I'm missing something about how this test works.

Reply:
Resolving this as you followed up offline.